src/sentry/api/endpoints/project_rules.py

```diff
@@ -128,6 +138,40 @@ def post(self, request: Request, project) -> Response:
             "frequency": data.get("frequency"),
             "user_id": request.user.id,
         }
+        matchers = [key for key in list(kwargs.keys()) if key not in ("name", "user_id")]
+        existing_rules = Rule.objects.filter(project=project, status=0)
+        for existing_rule in existing_rules:
+            keys = 0
+            matches = 0
+            for matcher in matchers:
+                if existing_rule.data.get(matcher) and kwargs.get(matcher):
```

Comment:
If, for example, you don't specifically select an environment the data won't be there